### PR TITLE
add option to run netdata in the background

### LINKED ISSUE #4295: Make netdata run in foreground by default
Current behavior is causing bad UX since it can display some log lines before forking. Which in turn can cause people to think that netdata crashed.

We also have an option `-D` which is counter-intuitive since it uses a negation and means "don't daemonize". It should be replaced by `-d` after making netdata by default running in foreground.

### PR DESCRIPTION
**Pull request by pohzipohzi**
fixes #4295

### CODE REVIEW

**Comment by ktsaou** [comment]
Hi,

thank you.

The way you have implemented this, a lot of installations out there will break in the next update.
So, you can't remove the `-D` option. It has to stay there, probably forever.
You need to add a new option `-d` for running netdata in background.

Then we can to the decision of which one will be the default.
Again, I don't want to break existing installations. If you check the `system` directory you will see that a lot of startup script don't use `-D`. So, they expect netdata to fork without any option. Also a lot of people start netdata via `rc.local` again without any option. All these will break.

So, I would add the `-d` option to indicate "run in background", but I wouldn't change the default, until we find a way to not break existing installations. There are a few ways to detect this (e.g. checking if the stdout is attached to a console, but these need some more work).

**Comment by ktsaou** [comment]
> Not forever. Just until we have full control over configuration and packaging and we release next major version.

I was talking about the `-D` option. Not about the default behavior. There is no point to remove an option. If can stay there, even if it just confirms the default and does nothing...

**Comment by paulfantom** [comment]
I am also talking about `-D` option. According to semantic versioning major release consist of breaking changes, so custom installations can fail and having control over default installation and upgrade process allows for easy transition in non-custom installations.


>  There is no point to remove an option. If can stay there, even if it just confirms the default and does nothing...

If option does nothing then it is dead code and that is the reason why it should be removed.

**Comment by ktsaou** [comment]
> According to semantic versioning major release consist of breaking changes, so custom installations can fail and having control over default installation and upgrade process allows for easy transition in non-custom installations.

I agree with one addition: semantic versioning major release consist of breaking changes **that are technically impossible to otherwise overcome**.

So, if it is technically possible to be backwards compatible, we should always prefer it over breaking backwards compatibility.

**Comment by ktsaou** [comment]
Of course we need to balance this. Sometimes it is technically possible to maintain backwards compatibility but it is just too much of a hassle to do, compared to the benefits it provides.

To my understanding, removal of the `-D` option will unnecessarily break existing installations. It costs us really nothing to keep it and will cost a lot to too many of our users to remove it.

**Comment by paulfantom** [comment]
> To my understanding, removal of the -D option will unnecessarily break existing installations.

We are not releasing 2.x version and till that I am not advocating on removal on `-D`. I will be when we will start preparing such release.

**Comment by pohzipohzi** [comment]
@ktsaou @paulfantom thank you for the feedback. I've readded the `-D` option. Does this work?

**Comment by ktsaou** [comment]
hm... it has more changes to it.
Could you please just add the `-d` option and leave everything else like it was?

File: daemon/main.c [hunk 1]
```diff
@@ -729,6 +730,8 @@ int main(int argc, char **argv) {
                 case 'D':
                     dont_fork = 1;
                     break;
+                case 'd':
+                    break;
```
Comment (ktsaou) [inline daemon/main.c]: and:

```
dont_fork = 0;
````

So, to change later the default we can just change the default value of `dont_fork`.

Comment (pohzipohzi) [reply daemon/main.c]: updated!

**Comment by ktsaou** [comment]
thank you!

**Pull request merged**